packages/peregrine/lib/talons/ForgotPassword/useForgotPassword.js

```diff
@@ -1,37 +1,53 @@
 import { useCallback, useState } from 'react';
+import { useMutation } from '@apollo/react-hooks';
 import { useUserContext } from '@magento/peregrine/lib/context/user';
 
 /**
  * Returns props necessary to render a ForgotPassword form.
- * @param {function} props.onClose callback function to invoke when closing the form
+ *
+ * @param {function} props.onCancel - callback function to call when user clicks the cancel button
+ * @param {function} props.mutations.resetPasswordMutation - mutation to call when the user clicks the submit button
+ *
+ * @returns {{
+ *  formErrors: [Error],
+ *  forgotPasswordEmail: string,
+ *  inProgress: boolean,
+ *  isResettingPassword: boolean,
+ *  handleCancel: function,
+ *  handleFormSubmit: function,
+ * }}
  */
 export const useForgotPassword = props => {
-    const [{ isResettingPassword }, { resetPassword }] = useUserContext();
+    const [{ isResettingPassword }] = useUserContext();
 
-    const { onClose } = props;
+    const { onCancel, mutations } = props;
 
     const [inProgress, setInProgress] = useState(false);
     const [forgotPasswordEmail, setForgotPasswordEmail] = useState(null);
 
+    const [resetPassword, { error: resetPasswordError }] = useMutation(
+        mutations.resetPasswordMutation
```

Comment:
I'd probably call this, or the other mutation that actually sets the new password, something else, just to avoid confusion :)